src/AppInstallerRepositoryCore/Manifest/YamlParser.cpp

```diff
@@ -197,13 +196,19 @@ namespace AppInstaller::Manifest
 
     std::vector<ValidationError> YamlParser::ParseManifest(const YAML::Node& rootNode, Manifest& manifest, bool fullValidation)
     {
+        // Detects empty files with a better error.
+        if (!rootNode.IsMap())
+        {
+            THROW_EXCEPTION_MSG(ManifestException(APPINSTALLER_CLI_ERROR_INVALID_MANIFEST), "The manifest does not contain a valid root.");

```

Comment:
>APPINSTALLER_CLI_ERROR_INVALID_MANIFEST [](start = 50, length = 39)

There's an existing error code for manifest failure APPINSTALLER_CLI_ERROR_MANIFEST_FAILED, should we merge the 2?

Reply:
That is the more generic parsing error.  It is rarely bad to have a more specific error.

---
In reply to: [468974896](https://github.com/Microsoft/pkg/pull/535#discussion_r468974896) [](ancestors = 468974896)